lib/cli/src/commands/run.rs

```diff
@@ -92,94 +90,154 @@ fn is_dir(e: &walkdir::DirEntry) -> bool {
     meta.is_dir()
 }
 
-impl RunWithoutFile {
-    /// Given a local path, returns the `Run` command (overriding the `--path` argument).
-    pub fn into_run_args(
-        mut self,
-        package_root_dir: PathBuf, // <- package dir
-        command: Option<&str>,
-        _debug_output_allowed: bool,
-    ) -> Result<Run, anyhow::Error> {
-        let (manifest, pathbuf) =
-            wasmer_registry::get_executable_file_from_path(&package_root_dir, command)?;
+#[derive(Debug, Parser, Clone, Default)]
+/// The options for the `wasmer run` subcommand
+pub struct Run {
+    /// File, URL or package to run
+    #[clap(name = "FILE", parse(try_from_str))]
+    pub(crate) path: SplitVersion,
 
-        #[cfg(feature = "wasi")]
-        {
-            let default = HashMap::default();
-            let fs = manifest.fs.as_ref().unwrap_or(&default);
-            for (alias, real_dir) in fs.iter() {
-                let real_dir = package_root_dir.join(&real_dir);
-                if !real_dir.exists() {
-                    if _debug_output_allowed {
-                        println!(
-                            "warning: cannot map {alias:?} to {}: directory does not exist",
-                            real_dir.display()
-                        );
-                    }
-                    continue;
-                }
+    /// Registry to run the FILE from (optional)
+    #[clap(short = 'r', long, name = "registry")]
+    pub(crate) registry: Option<String>,
```

Comment:
Why whould we have a separate registry flag if one can provide a URL?

Reply:
Because we might want to install a package from `wapm.dev` / `wapm.io` instead of going through `wasmer login` / `wapm config set registry.url`

Reply:
But that would just be `https://wapm.dev/ns/name` then?

Reply:
Downloading a package doesn't require login.

Reply:
It is useful if you want to specify the registry for a package, but you don't want to mess with the `wapm.config`. Ex. by default the default registry in prod is `wapm.io`, but the test are always locked to `wapm.dev`, since some packages don't exist on .io.

`wasmer run {url}` only downloads + runs webc files, not tar.gz packages.

I also don't think you can specify versions yet like `/name@version`

For now I'd like to leave it as is, at least until we've migrate the website documentation from `wapm run` to `wasmer run`.